src/os/FileStore.cc

```diff
@@ -4180,7 +4196,8 @@ int FileStore::_setattrs(coll_t cid, const ghobject_t& oid, map<string,bufferptr
 	continue;
     }
     omap_remove.insert(p->first);
-    inline_set.insert(*p);
+    if (!inline_set.count(p->first))
+      inline_set.insert(make_pair(p->first, make_pair(p->second, -1)));
```

Comment:
@liewegas , if the pool has not snapshot, the snapset attr should only have fixed size and only one chuink?
And the pool has not snapshot and not watchers for specially object, the object would also has fixed size and only one chunk after https://github.com/ceph/ceph/pull/6510.
if the above is true, I think _setattrs could not go through this?
I think it is worthwhile while most production environment has not snapshot?